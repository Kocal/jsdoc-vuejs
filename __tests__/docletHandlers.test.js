const path = require('path')

const docletHandlers = require('./../lib/docletHandlers')

const createJSDocEvent = (filename) => {
  return {
    doclet: {
      meta: {
        filename
      },
      description: ''
    }
  }
}

const createJSDocEventForMethod = (filename) => {
  const event = createJSDocEvent(filename)

  event.doclet.kind = 'function'

  return event
}

describe('docletHandlers', () => {
  describe('BetterCounter', () => {
    const filename = path.join(__dirname, '../example/src/BetterCounter.vue')
    const BetterCounter = require(filename)
    const JSDocEvent = createJSDocEvent(filename)

    describe('handleComputed', () => {
      it('should properly handle event\'s doclet', () => {
        docletHandlers.handleComputed(JSDocEvent, BetterCounter)

        expect(JSDocEvent.doclet.description).toMatchSnapshot()
      })
    })

    describe('handleData', () => {
      it('should properly handle event\'s doclet', () => {
        docletHandlers.handleData(JSDocEvent, BetterCounter)

        expect(JSDocEvent.doclet.description).toMatchSnapshot()
      })
    })

    describe('handleMethods', () => {
      it('should properly handle event\'s doclet', () => {
        docletHandlers.handleMethods(JSDocEvent)

        expect(JSDocEvent.doclet.description).toMatchSnapshot()
      })
    })

    describe('handleProps', () => {
      it('should properly handle event\'s doclet', () => {
        docletHandlers.handleProps(JSDocEvent, BetterCounter)

        expect(JSDocEvent.doclet.description).toMatchSnapshot()
      })
    })
  })

  describe('EmptyComponent', () => {
    const filename = path.join(__dirname, 'fixtures/EmptyComponent.vue')
    const EmptyComponent = require(filename)
    const JSDocEvent = createJSDocEvent(filename)

    describe('handleComputed', () => {
      it('should properly handle event\'s doclet', () => {
        docletHandlers.handleComputed(JSDocEvent, EmptyComponent)

        expect(JSDocEvent.doclet.description).toMatchSnapshot()
      })
    })

    describe('handleData', () => {
      it('should properly handle event\'s doclet', () => {
        docletHandlers.handleData(JSDocEvent, EmptyComponent)

        expect(JSDocEvent.doclet.description).toMatchSnapshot()
      })
    })

    describe('handleMethods', () => {
      it('should properly handle event\'s doclet', () => {
        docletHandlers.handleMethods(JSDocEvent)

        expect(JSDocEvent.doclet.description).toMatchSnapshot()
      })
    })

    describe('handleProps', () => {
      it('should properly handle event\'s doclet', () => {
        docletHandlers.handleProps(JSDocEvent, EmptyComponent)

        expect(JSDocEvent.doclet.description).toMatchSnapshot()
      })
    })
  })

  describe('methods and hooks', () => {
    // this case should not event exists
    it('should not modify doclet if not vue file', () => {
      const event = createJSDocEventForMethod('foo.bar')
      const finalEvent = createJSDocEventForMethod('foo.bar')

      expect(event).toEqual(finalEvent)
      docletHandlers.handleMethods(finalEvent)
      expect(event).toEqual(finalEvent)
    })

    it('should modify name when it\'s a hook', () => {
      const event = createJSDocEventForMethod('foo.vue')

      event.doclet.name = 'created'
      docletHandlers.handleMethods(event)
      expect(event.doclet.name).toEqual('[Hook] created')
    })

    it('should not modify name when it\'s not a hook', () => {
      const event = createJSDocEventForMethod('foo.vue')

      event.doclet.name = 'not_a_hook'
      docletHandlers.handleMethods(event)
      expect(event.doclet.name).toEqual('not_a_hook')
    })
  })
})
