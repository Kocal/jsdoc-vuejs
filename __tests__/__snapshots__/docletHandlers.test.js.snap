// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docletHandlers BetterCounter handleComputed should properly handle event's doclet 1`] = `"<h3 class=\\"subsection-title\\">Computed</h3><table><thead><tr><th>Name</th></tr></thead><tbody><tr><td>message</td></tr></tbody></table>"`;

exports[`docletHandlers BetterCounter handleData should properly handle event's doclet 1`] = `"<h3 class=\\"subsection-title\\">Computed</h3><table><thead><tr><th>Name</th></tr></thead><tbody><tr><td>message</td></tr></tbody></table><h3 class=\\"subsection-title\\">Data</h3><table><thead><tr><th>Name</th></tr></thead><tbody><tr><td>counter</td></tr></tbody></table>"`;

exports[`docletHandlers BetterCounter handleMethods should properly handle event's doclet 1`] = `"<h3 class=\\"subsection-title\\">Computed</h3><table><thead><tr><th>Name</th></tr></thead><tbody><tr><td>message</td></tr></tbody></table><h3 class=\\"subsection-title\\">Data</h3><table><thead><tr><th>Name</th></tr></thead><tbody><tr><td>counter</td></tr></tbody></table>"`;

exports[`docletHandlers BetterCounter handleProps should properly handle event's doclet 1`] = `"<h3 class=\\"subsection-title\\">Computed</h3><table><thead><tr><th>Name</th></tr></thead><tbody><tr><td>message</td></tr></tbody></table><h3 class=\\"subsection-title\\">Data</h3><table><thead><tr><th>Name</th></tr></thead><tbody><tr><td>counter</td></tr></tbody></table><h3 class=\\"subsection-title\\">Props</h3><table><thead><tr><th>Name</th><th>Type</th><th>Required</th><th>Default value</th></tr></thead><tbody><tr><td>initialCounter</td><td>Number</td><td><b>true</b></td><td></td></tr><tr><td>step</td><td>Number</td><td>false</td><td>1</td></tr></tbody></table>"`;

exports[`docletHandlers EmptyComponent handleComputed should properly handle event's doclet 1`] = `"<h3 class=\\"subsection-title\\">Computed</h3><p>No computed data to display.</p>"`;

exports[`docletHandlers EmptyComponent handleData should properly handle event's doclet 1`] = `"<h3 class=\\"subsection-title\\">Computed</h3><p>No computed data to display.</p><h3 class=\\"subsection-title\\">Data</h3><p>No data to display.</p>"`;

exports[`docletHandlers EmptyComponent handleMethods should properly handle event's doclet 1`] = `"<h3 class=\\"subsection-title\\">Computed</h3><p>No computed data to display.</p><h3 class=\\"subsection-title\\">Data</h3><p>No data to display.</p>"`;

exports[`docletHandlers EmptyComponent handleProps should properly handle event's doclet 1`] = `"<h3 class=\\"subsection-title\\">Computed</h3><p>No computed data to display.</p><h3 class=\\"subsection-title\\">Data</h3><p>No data to display.</p><h3 class=\\"subsection-title\\">Props</h3><p>No props to display.</p>"`;
